zebra-network/src/constants.rs

```diff
@@ -258,7 +261,10 @@ mod tests {
         assert!(EWMA_DEFAULT_RTT > REQUEST_TIMEOUT,
                 "The default EWMA RTT should be higher than the request timeout, so new peers are required to prove they are fast, before we prefer them to other peers.");
 
-        assert!(EWMA_DECAY_TIME > REQUEST_TIMEOUT,
+        let request_timeout_nanos = REQUEST_TIMEOUT.as_secs_f64()
+            + f64::from(REQUEST_TIMEOUT.subsec_nanos()) * NANOS_PER_SECOND;
```

Comment:
Likely Bug: The seconds need to be multiplied by `NANOS_PER_SECOND` to yield nanoseconds.

```suggestion
        let request_timeout_nanos = REQUEST_TIMEOUT.as_secs_f64() * NANOS_PER_SECOND
            + f64::from(REQUEST_TIMEOUT.subsec_nanos());
```